src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
```

Comment:
The use of `cache.containsKey(userId)` followed by `cache.get(userId)` can be optimized by using `cache.computeIfAbsent(userId, key -> ...)` to avoid double lookup in the map.